secretsstoresynccontroller/config/admissionpolicies/create_update_specific_types.yaml

```diff
@@ -0,0 +1,32 @@
+# This policy is used to deny the creation or update of secrets of types other than Opaque, 
+# kubernetes.io/basic-auth, kubernetes.io/ssh-auth and kubernetes.io/tls
+# TODO: Use parameters to allow the user to specify the name of the SecretSyncController.
+apiVersion: admissionregistration.k8s.io/v1beta1
+kind: ValidatingAdmissionPolicy
+metadata:
+  name: "sscontroller-create-update-secrets-types-policy"
+spec:
+  failurePolicy: Fail
+  matchConstraints:
+    resourceRules:
+    - apiGroups:   [""]
+      apiVersions: ["v1"]
+      operations:  ["CREATE","UPDATE"]
+      resources:   ["secrets"]
+  variables:
+  - name: typeCheck
+    expression: "(object.type == 'Opaque' || object.type == 'kubernetes.io/basic-auth' || object.type == 'kubernetes.io/ssh-auth' || object.type == 'kubernetes.io/tls')"
```

Comment:
if we want to hard-code something, maybe just hard-code forbidding service account token secrets

for the things we want to allow, consider how to avoid this being hard-coded, maybe with use of parameters (like a configmap params object) or _maaaaaaybe_ via an authz check (maybe check against a virtual subresource like secrets/type and have an rbac role that allows the types we expect)